lib/def.js

```diff
@@ -173,6 +181,20 @@
           return this.base
         }
         return new infer.Arr(types || inner)
+      } else if (this.eat("{")) {
+        var types = [], names = [], computed = false
+        if (!this.eat("}")) {
+          computed = this.parseList(comp, types, names)
+          this.eat("}") || this.error()
+        }
+        if (computed) {
+          this.error();
```

Comment:
You'll have to write a `computedObject` function analogous to `computedArray`, an call that. These are for return types that depend on the function's input type (i.e. you could say `fn(arg: ?) -> {content: !0}` for a function that wraps its argument in an object.